### DISCUSSION TITLE
Best practice for monorepo UI library?

### DISCUSSION THREAD

**Discussion by limegorilla** [tailwindlabs/tailwindcss#9890] [Help] [answered]
So I have a monorepo, `website` - it's a turborepo with roughly the following structure:

```
./
  apps/
    site1/
      postcss.config.js
      {...}
    site2/
    {...}
  packages/
    ui/
      exampleComponent.tsx
      postcss.config.js
      {...}
    {...}
  tailwind.config.js
  {...}
```

And all of my `postcss.config.js` contains:
```js
module.exports = {
	plugins: {
		tailwindcss: { config: '../../tailwind.config.js' },
		autoprefixer: {},
	},
};
```

Essentially what I want is one tailwind configuration. All of the sites will use the same UI, however some elements - like branding and colours - will change. I'm using tailwind's CSS variables to be able to apply those to my config - my `theme.colors` looks like this:
```js
/** @type {import('tailwindcss').Config} */

// Imports
const colors = require("tailwindcss/colors");

module.exports = {
	content: [
		"./packages/ui/common/nav/NavBar.tsx",
		"./**/pages/**/*.{js,ts,jsx,tsx}",
		"./**/app/**/*.{js,ts,jsx,tsx}",
		"./**/components/**/*.{js,ts,jsx,tsx}",
	],
	theme: {
		colors: {
			// Custom colours - set on project.
			primary: "rgb(var(--color-primary))",
			secondary: "rgb(var(--color-secondary))",
			tertiary: "rgb(var(--color-tertiary))",
```

I need to be able to have the colour variables from the site come through to the package.

Is there any way to do this? Am I setting this up the right way?

This currently isn't working at at all - using `next.js ./app` means I have a server error, put using `./pages` comes back with this:
```
Module parse failed: Unexpected token (3:8)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export default function NavBar() {
|     return (
>         <nav className="w-screen ml-2 bg-black h-4">
|             hellodd 
|         </nav>
```

**Comment by MattHeslington** [comment, marked as answer]
There's a tutorial [here](https://shubhamverma.me/blog/setup-tailwind-css-with-turborepo) I have used for Turborepo and works fine. Basically, install Tailwind as a package, and link to the package tailwind.config.js file and postcss.config.js files in your apps and ui directory through :

// postcss.config.js
```
module.exports = require("config/postcss.config")
```

and 

// tailwind.config.js
```
module.exports = require("config/tailwind.config")
```

**Reply by neznayer** [reply]
This a nice tutorial! I've come up with pretty similar pattern. It is rendering properly, when I use utility class defined in some of the `apps` since it is getting transpiled inside that app.
But the problem I've encounter now, is that when editing `ui` components, the Intellisense has no idea about tailwind config in the `apps/{name}`.
